homeassistant/components/jewish_calendar/service.py

```diff
@@ -0,0 +1,341 @@
+"""Services to return HDate instances & other information."""
+
+from __future__ import annotations
+
+import datetime as dt
+import logging
+from typing import cast
+
+from hdate import HDate, HebrewDate, Zmanim, htables
+import voluptuous as vol
+
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+import homeassistant.util.dt as dt_util
+from homeassistant.util.json import JsonObjectType
+
+DOMAIN = "jewish_calendar"
+
+LOGGER = logging.getLogger(__name__)
+# Maps Hebrew & English month names to Months enum values.
+MONTH_INPUT_MAP = {
+    name: htables.Months(index + 1)
+    for index, m in enumerate(htables.MONTHS)
+    # Require users to specify which Adar they want in case of leap year.
+    if m.english != "Adar"
+    for name in (m.english, m.hebrew)
+}
+# Maps Hebrew & English holiday names to Holiday)
+HOLIDAY_INPUT_MAP = {
+    name: holiday
+    for holiday in htables.HOLIDAYS
+    for name in (
+        holiday.name,
+        holiday.description.english,
+        holiday.description.hebrew.long,
+        holiday.description.hebrew.short,
+    )
+    if name
+}
+
+OUTPUT_SCHEMA = {
+    vol.Required("include_hebrew_date_info", default=False): cv.boolean,
+    vol.Required("include_holiday_info", default=False): cv.boolean,
+    vol.Required("include_zmanim", default=False): cv.boolean,
+}
+
+
+class JewishCalendarServices:
```

Comment:
Why is this a class? It looks like there's no need for this, if that's the case change to regular functions instead.

Reply:
I store the integration's settings in `self`, and consume them in `self._build_response` and `self.resolve_date_param()`

Reply:
Changed to an `async_setup_services()` closure